core/src/main/java/nl/nn/adapterframework/stream/MessageContext.java

```diff
@@ -42,24 +50,63 @@ public class MessageContext extends LinkedHashMap<String,Object> {
 	public static final String METADATA_LOCATION = "Metadata.Location";
 	public static final String METADATA_MIMETYPE = "Metadata.MimeType";
 
+	private Map<String, Serializable> data = new LinkedHashMap<>();
+
 	public MessageContext() {
 		super();
 	}
 	public MessageContext(String charset) {
 		this();
 		withCharset(charset);
 	}
-	public MessageContext(Map<String,Object> base) {
+	public MessageContext(Map<String, ?> base) {
 		this();
 		withAllFrom(base);
 	}
 
-	public MessageContext withAllFrom(Map<String,Object> base) {
+	public MessageContext(MessageContext base) {
+		this(base.data);
+	}
+
+	public MessageContext withAllFrom(Map<String,?> base) {
 		if (base!=null) {
 			putAll(base);
 		}
 		return this;
 	}
+
+	public void putAll(Map<String, ?> base) {
+		if (base!=null) {
+			base.forEach((key, value) -> {
+				if (value instanceof Serializable) data.put(key, (Serializable) value);
```

Comment:
En anders? Ten minste een log melding denk ik, of als transient toevoegen oid.

Reply:
Goed punt. Ik denk dat we in de context wel een map van `Map<String, Object>` moeten houden maar dat we in `writeObject()` dan alles eruit moeten filteren wat niet `Serializable` is, zodat er dan in ieder geval geen onverwachtte fouten optreden bij het schrijven naar de error-store, en op dat moment een warning loggen voor al die keys.